smithy-model/src/test/java/software/amazon/smithy/model/loader/LoaderVisitorTest.java

```diff
@@ -1,209 +0,0 @@
-    private static final TraitFactory FACTORY = TraitFactory.createServiceFactory();
-
-    @Test
-    public void callingOnEndTwiceIsIdempotent() {
-        LoaderVisitor visitor = new LoaderVisitor(FACTORY);
-
-        assertThat(visitor.onEnd(), is(visitor.onEnd()));
-    }
-
-    @Test
-    public void cannotDuplicateTraitDefs() {
-        Assertions.assertThrows(SourceException.class, () -> {
-            LoaderVisitor visitor = new LoaderVisitor(FACTORY);
-            StringShape def1 = StringShape.builder()
-                    .id("foo.baz#Bar")
-                    .addTrait(TraitDefinition.builder().build())
-                    .build();
-            StringShape def2 = StringShape.builder()
-                    .id("foo.baz#Bar")
-                    .addTrait(TraitDefinition.builder().selector(Selector.parse("string")).build())
-                    .build();
-
-            visitor.onShape(def1);
-            visitor.onShape(def2);
-            visitor.onEnd();
-        });
-    }
-
-    @Test
-    public void ignoresDuplicateTraitDefsFromPrelude() {
-        LoaderVisitor visitor = new LoaderVisitor(FACTORY);
-        Shape def1 = StructureShape.builder()
-                .id("smithy.api#deprecated")
-                .addTrait(TraitDefinition.builder().build())
-                .build();
-        Shape def2 = StructureShape.builder()
-                .id("smithy.api#deprecated")
-                .addTrait(TraitDefinition.builder().build())
-                .build();
-
-        visitor.onShape(def1);
-        visitor.onShape(def2);
-        List<ValidationEvent> events = visitor.onEnd().getValidationEvents();
-
-        assertThat(events, empty());
-    }
-
-    @Test
-    public void cannotDuplicateNonPreludeTraitDefs() {
-        Assertions.assertThrows(SourceException.class, () -> {
-            LoaderVisitor visitor = new LoaderVisitor(FACTORY);
-            Shape def1 = StructureShape.builder()
-                    .id("smithy.example#deprecated")
-                    .addTrait(TraitDefinition.builder().build())
-                    .build();
-            Shape def2 = StructureShape.builder()
-                    .id("smithy.example#deprecated")
-                    .addTrait(TraitDefinition.builder().build())
-                    .build();
-
-            visitor.onShape(def1);
-            visitor.onShape(def2);
-            visitor.onEnd();
-        });
-    }
-
-    @Test
-    public void cannotDuplicateTraits() {
-        LoaderVisitor visitor = new LoaderVisitor(FACTORY);
-        ShapeId id = ShapeId.from("foo.bam#Boo");
-        visitor.onShape(StringShape.builder().id(id));
-        visitor.onTrait(id, DocumentationTrait.ID, Node.from("abc"));
-        visitor.onTrait(id, DocumentationTrait.ID, Node.from("def"));
-        List<ValidationEvent> events = visitor.onEnd().getValidationEvents();
-
-        assertThat(events, not(empty()));
-    }
-
-    @Test
-    public void createsDynamicTraitWhenTraitFactoryReturnsEmpty() {
```

Comment:
This test isn't carried over even though the functionality is retained and would be a break if the behavior changed unexpectedly.

Reply:
Good catch. Fixed.